lib/fv/api_resource.rb

```diff
@@ -61,24 +61,48 @@ def self.define_attribute_readers(*attrs)
     end
 
     def self.has_many(*args)
+      memoize_relationships(*args) do |relationship, to_resource_class|
+        FV::HasManyAssociation.new(
+          self,
+          to_resource_class,
+          relationship
+        ).tap do |association|
+          @associations << association
+        end
+      end
+    end
+
+    def self.belongs_to(*args)
+      memoize_relationships(*args) do |relationship, to_resource_class|
+        to_resource_class.new(
+          self.class.client.request(:get, "#{path}/#{relationship}").data
+        )
+      end
+    end
+
+    def self.memoize_relationships(*args, &block)
       args.each do |relationship|
         define_method(relationship) do
           key = "@#{relationship}"
           return instance_variable_get(key) if instance_variable_defined?(key)
 
-          module_name = self.class.to_s.split('::')[0..-2].join('::')
-          to_resource_classname = relationship.to_s.singularize.camelize
-          association = FV::HasManyAssociation.new(
-            self,
-            "#{module_name}::#{to_resource_classname}".constantize,
-            relationship
+          value = instance_exec(
+            relationship,
+            resource_class_for_relationship(relationship),
+            &block
           )
-          @associations << association
-          instance_variable_set(key, association)
+
+          instance_variable_set(key, value)
         end
       end
     end
 
+    def resource_class_for_relationship(relationship)
+      module_name = self.class.to_s.split('::')[0..-2].join('::')
```

Comment:
this seems very dirty...I wonder if there is a better way to do this?  Why not use `self.superclass`?

Reply:
The purpose of this is to do resource discovery within the consuming code's namespace. This grabs the namespace of the current resource, and searches in that namespace for the related resource.

`self.class.superclass` would always return `FV::ApiResource`, which isn't what I want.